### DISCUSSION TITLE
Make useControllableState ready for public usage

### DISCUSSION THREAD

**Discussion by misha-erm** [radix-ui/primitives#1146] [Ideas]
Hello 👋🏻 

I'm in a situation when I want to have similar optional Controlled/Uncontrolled behaviour on my wrapper around radix-ui/primitives.
I've found this package https://github.com/radix-ui/primitives/tree/main/packages/react/use-controllable-state but it says it's only for internal usage. 

What do you think about making it available for public usage and what can be downsides of using it in public right now? 

Thank you 🙏🏻

**Comment by jjenzz** [comment]
We are reluctant to release our hooks library to the public as it adds an additional "support" requirement to our workload. Currently we can change these utilities relatively freely without worrying about impact on consumers.

Feel free to create your own hooks from ours though if you like, given that we are MIT licensed.
